### PR TITLE
[viewchange] remove redundant viewID

### PR DESCRIPTION
**Pull request by LeoHChen**
Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -77,7 +77,6 @@ type Consensus struct {
 	// blockNum: the next blockNumber that FBFT is going to agree on,
 	// should be equal to the blockNumber of next block
 	blockNum uint64
-	viewID   uint64
```
Comment (rlan35) [inline consensus/consensus.go:80]: This view ID is not redundant. This is the viewID for the last successful consenus. The current.viewID is the moving viewID during view change. They can not be merged together, at least based on the current logic. Things will break if we don't keep track of them separately in current code.

Comment (LeoHChen) [reply consensus/consensus.go:80]: wow! that's a precious hidden knowledge. I can see why it is very confusing.

Comment (rlan35) [reply consensus/consensus.go:80]: yup. The whole view change code logic is not clean. Very error-prone.

**Comment by LeoHChen** [comment]
this PR is replaced by https://github.com/harmony-one/harmony/pull/3347

**Pull request closed without merging**